staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -87,7 +87,7 @@ type Config struct {
 	AuthConfigPersister AuthProviderConfigPersister
 
 	// Exec-based authentication provider.
-	ExecProvider *clientcmdapi.ExecConfig
```

Comment:
rest.Config is probably one of our most heavily used structs... is there a way to do this without breaking existing call sites (e.g. add `ClusterConfig runtime.Object` to `ExecConfig`, or add a peer `ExecClusterConfig runtime.Object` field)?

Reply:
While I agree that `rest.Config` is heavily used, I doubt the `ExecProvider` field is often used.  I purposefully broke the call site so that it would be easier for folks who actually use the `ExecProvider` functionality to notice that something changed.  The [custom consumers](https://grep.app/search?q=.ExecProvider&case=true) of this field should really be notified that "hey, you are no longer consuming this struct correctly."

Reply:
> The [custom consumers](https://grep.app/search?q=.ExecProvider&case=true) of this field should really be notified that "hey, you are no longer consuming this struct correctly."

Someone manually constructing a rest.Config without specifying cluster-specific config info should be able to leave the new field set to nil, right? How would that not be correct?

Reply:
> Someone manually constructing a rest.Config without specifying cluster-specific config info should be able to leave the new field set to nil, right?

Sure but I am arguing that the set of folks constructing a `rest.Config` with an `exec` section is very small.  Since they leave `ExecProvider` unset, this change will not cause them any issue.  This PR covers the internals of `client-go` which needed to be updated.

The only folks that we will annoy are those that have a static `exec` config that must now be slightly moved in the struct declaration.  I was able to find [one example](https://github.com/argoproj/argo-cd/blob/24acaefce38545a5d39b2cbdeadd453a646e4df3/pkg/apis/application/v1alpha1/types.go#L2274-L2278).

> How would that not be correct?

I would argue that all of these `rest.Config` consumers are subtly broken because they assume that all the information they need is contained within the `ExecProvider` struct: [[1]](https://github.com/argoproj/gitops-engine/blob/7d3da9f16e353692824a6e83fc31de113b6a9251/pkg/utils/kube/kube.go#L265) [[2]](https://github.com/kudobuilder/kuttl/blob/5c4dbec082841d542b085e9f5ad3bc4a17caaece/pkg/test/utils/kubernetes.go#L1092-L1106) [[3]](https://github.com/int128/kauthproxy/blob/c2cffe513e430d74645db76d78308ab73e0bf6da/pkg/adaptors/transport/transport.go#L29-L37) [[4]](https://github.com/VirtusLab/kubedrainer/blob/91e4de8ebf63724b37c18e5097888fe6e263537b/pkg/kubernetes/client.go#L39).  These are special in the sense that they seem to be parsing the `rest.Config` in an attempt to understand it and act on its contents instead of just passing it to `client-go`.